File: src/main/java/br/com/pnp/compiler/model/construct/type/primitives/String.java
package br.com.pnp.compiler.model.construct.type.primitives;

import br.com.pnp.compiler.model.construct.type.Type;
import br.com.pnp.compiler.model.expression.Expression;

class String extends Type {
    @Override
    public boolean isTypeOf(Object value) {
        if (value instanceof java.lang.String) {
            return true;
        }
        if (value instanceof String) {
            return true;
        }
        if (value instanceof Expression) {
            return ((Expression) value).getType() instanceof String;
        }
        return false;
    }

    @Override
    public boolean isPrimitive() {
        return true;
    }

    @Override
    public java.lang.String toString() {
        return "string";
    }
}

File: src/main/kotlin/br/com/pnp/model/construct/type/primitive/String.kt
package br.com.pnp.model.construct.type.primitive

import br.com.pnp.model.construct.type.Type

class String : Type() {
    override val isPrimitiveType = true

    override fun isTypeOf(value: Any): kotlin.Boolean {
        if (value is kotlin.String) {
            return true
        }
        if (value is String) {
            return true
        }
//        return if (value is Expression) {
//            (value as Expression).getType() is String
//        }
        return false
    }
}
